test/models/name_test.rb

```diff
@@ -3284,6 +3284,31 @@ def test_genus_of_synonym
     assert_names_equal(names(:stereum), names(:stereum_hirsutum).accepted_genus)
   end
 
+  def test_can_propagate
+    assert(names(:coprinus).can_propagate?,
+           "Classification of genus Names should be propagable")
+
+    assert_false(names(:coprinus_sensu_lato).can_propagate?,
+                 "Classification of Names sensu lato should not be propagable")
+    [:eukarya, :fungi, :ascomycota, :ascomycetes, :agaricales,
+     :agaricaceae].each do |name|
+      assert_false(names(name).can_propagate?,
+                   "Classification of Names above genus should be propagable")
+    end
+    assert_false(
+      names(:amanita_subgenus_lepidella).can_propagate?,
+      "Classification of infra-generic Names should not be propagable"
+    )
+    assert_false(
+      names(:amanita_boudieri_var_beillei).can_propagate?,
+      "Classification of infra-specific Names should not be propagable"
+    )
+    assert_false(
+      names(:boletus_edulis_group).can_propagate?,
+      "Classification of group or clade Names should not be propagable"
+    )
```

Comment:
Dry and simply this. Note type in original should be "should _not_ be propagated"
Something like:
```ruby
    [<all the bad name symbols>].each do |name|
      assert_false(names(name).can_propagate?,
                   "Classification of Names of rank {#name.ran} should not propagate")
    end
```
Might need to fixed pluralization of "genus" and "subgenus":
```ruby
irb(main):002> "genus".pluralize
=> "genus"
irb(main):006> "subgenus".pluralize
=> "subgenus"
```